### PR TITLE
Add popup_checkout task

### PR DESCRIPTION
**Pull request by Roophee**
### HTML & CSS practice: Hooli-style Popup

[Demo](https://roophee.github.io/) |
[Code base](https://github.com/Roophee/frontend-2021-homeworks/tree/popup_checkout_Roophee/submissions/Roophee/Popup_checkbox)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

File: submissions/Roophee/Popup_checkbox/base.html [hunk 1]
```diff
@@ -0,0 +1,483 @@
+<!DOCTYPE html>
+<html lang="ru-RU">
```
Comment (lenadgit) [inline submissions/Roophee/Popup_checkbox/base.html:2]: the language should accord to the content, could you please change to "en"

**Review by lenadgit — CHANGES_REQUESTED** [review]

File: submissions/Roophee/Popup_checkbox/css/main.css [hunk 2]
```diff
@@ -0,0 +1,821 @@
...
+
+.content-container {
+  background-color: #227493;
+  -ms-grid-row: 2;
+  -ms-grid-column: 1;
+  grid-area: content;
+  overflow: auto;
+  height: 800px;
+}
+/*# sourceMappingURL=main.css.map */
```
Comment (lenadgit) [inline submissions/Roophee/Popup_checkbox/css/main.css:821]: could you please add an empty lane at the end of the file?

**Review by lenadgit — CHANGES_REQUESTED** [review]

**Comment by lenadgit** [comment]
Great job! "grids" are welcome

File: submissions/Roophee/Popup_checkbox/css/main.css (hunk 2 above)
Comment (Roophee) [reply submissions/Roophee/Popup_checkbox/css/main.css]: I think, I did it. Please, make review again. Thanks!

Comment (Roophee) [reply submissions/Roophee/Popup_checkbox/css/main.css]: I mean , it was done after your review. Thanks).

File: submissions/Roophee/Popup_checkbox/css/main.scss [hunk 3]
```diff
@@ -0,0 +1,422 @@
...
+    justify-content: center;
+    align-content: center;
+}
+
+.nav-list {
+    list-style: none;
+    padding: 0;
+    margin: 0;
+    display: grid;
+    grid-template-columns: 60px 60px 60px 60px 60px;
```
Comment (A-Ostrovnyy) [inline submissions/Roophee/Popup_checkbox/css/main.scss:92]: For cases like this use the `repeat()` function, it makes your code more readable. 
[Read more](https://developer.mozilla.org/en-US/docs/Web/CSS/repeat()) about repeat function

File: submissions/Roophee/Popup_checkbox/index.html [hunk 4]
```diff
@@ -0,0 +1,484 @@
...
+
+                                                </symbol>
+
+                                            </svg>
+                                            <svg class="popup__item-icon" xmlns="http://www.w3.org/2000/svg">
+                                                <use xlink:href="#document">
+                                                </use>
+                                            </svg>
+                                        </a>
+                                        <span>Document</span>
```
Comment (A-Ostrovnyy) [inline submissions/Roophee/Popup_checkbox/index.html:278]: Avoid behavior when content bigger than a block with fixed sizes. If you don't control content use a different strategy for avoiding this.
image

File: submissions/Roophee/Popup_checkbox/index.html [hunk 5]
```diff
@@ -0,0 +1,484 @@
...
+
+                                            </svg>
+                                            <svg class="popup__item-icon popup__item-icon--more-button "
+                                                xmlns="http://www.w3.org/2000/svg">
+                                                <use xlink:href="#more1">
+                                                </use>
+                                            </svg>
+
+                                        </label>
+                                        <input type="checkbox" name="popup__checkbox2" class="popup__checkbox2-check"
```
Comment (A-Ostrovnyy) [inline submissions/Roophee/Popup_checkbox/index.html:302]: Watch the [video](https://www.youtube.com/watch?v=E6kLaaQFctU&t=413s) about hiding checkboxes in the right way.

File: submissions/Roophee/Popup_checkbox/css/main.scss (hunk 3 above)
Comment (Roophee) [reply submissions/Roophee/Popup_checkbox/css/main.scss:92]: Thanks, I will. "Per aspera ad astra!"

File: submissions/Roophee/Popup_checkbox/css/main.scss [hunk 6]
```diff
@@ -0,0 +1,422 @@
...
+    &:hover {
+        color: $hover-color;
+        font-size: 1.18rem;
+    }
+
+    &:active {
+        font-size: 0.9rem;
+    }
+
+    &:focus-within {
```
Comment (A-Ostrovnyy) [inline submissions/Roophee/Popup_checkbox/css/main.scss:112]: Focus indicates to the user about current focused element. Avoid couple focused elements it confusing users.
[Watch video](https://share.vidyard.com/watch/JwmjdxmE3nMUHETAg7Pt7T?)

File: submissions/Roophee/Popup_checkbox/css/main.scss [hunk 7]
```diff
@@ -0,0 +1,422 @@
...
+body {
+    width: 100%;
+    height: 100%;
+    padding: 0;
+    margin: 0;
+    font-family: Arial, Helvetica, sans-serif;
+    scroll-behavior: smooth;
+}
+
+.container {
```
Comment (A-Ostrovnyy) [inline submissions/Roophee/Popup_checkbox/css/main.scss:54]: On fullHD screen I have that white space at the bottom of the page, Please fix it

image

File: submissions/Roophee/Popup_checkbox/index.html [hunk 8]
```diff
@@ -0,0 +1,484 @@
...
+                                    <use xlink:href="#some">
+                                    </use>
+                                </svg>
+                            </label>
+
+                            <div class="popup__hide-pannel1">
+                                <ul class="popup__items-list1">
+                                    <li class="popup__item">
+                                        <a href="#">
+                                            <svg display="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (A-Ostrovnyy) [inline submissions/Roophee/Popup_checkbox/index.html:55]: Why you use two `svg` elements in every list item?

**Review by A-Ostrovnyy — CHANGES_REQUESTED** [review]
Good work!
In future PR add only source files you wrote/changed: no need to add images , sourcemap files, compiled files etc.

File: submissions/Roophee/Popup_checkbox/css/main.scss (hunk 3 above)
Comment (Roophee) [reply submissions/Roophee/Popup_checkbox/css/main.scss]: Thanks.  I made revisions.

File: submissions/Roophee/Popup_checkbox/index.html (hunk 4 above)
Comment (Roophee) [reply submissions/Roophee/Popup_checkbox/index.html]: I got it, thanks. Сorrected font-size in element.

File: submissions/Roophee/Popup_checkbox/index.html (hunk 5 above)
Comment (Roophee) [reply submissions/Roophee/Popup_checkbox/index.html]: That was my problem. I didn`t understand how to arrange elements and selectors. But I did it. Thanks.

File: submissions/Roophee/Popup_checkbox/css/main.scss (hunk 6 above)
Comment (Roophee) [reply submissions/Roophee/Popup_checkbox/css/main.scss:112]: Focus problem is gone.  I didn`t understand how to arrange elements and selectors. Fix this problem. Thanks.

File: submissions/Roophee/Popup_checkbox/css/main.scss (hunk 7 above)
Comment (Roophee) [reply submissions/Roophee/Popup_checkbox/css/main.scss:54]: Fixed. Set proper height of container. Thank you for your comment.

File: submissions/Roophee/Popup_checkbox/index.html (hunk 8 above)
Comment (Roophee) [reply submissions/Roophee/Popup_checkbox/index.html]: Sorry, is it wrong? I saw one educational video on youtube about sprites. Lector did like that. And I copy his approach. 
This my bad. I`ll learn this more thoroughly.
Thanks.

File: submissions/Roophee/Popup_checkbox/index.html (hunk 5 above)
Comment (A-Ostrovnyy) [reply submissions/Roophee/Popup_checkbox/index.html]: [Read more](https://developer.mozilla.org/en-US/docs/Web/CSS/:focus-within) about how `:focus-within` works.

File: submissions/Roophee/Popup_checkbox/index.html (hunk 4 above)
Comment (A-Ostrovnyy) [reply submissions/Roophee/Popup_checkbox/index.html]: Or add paddings to the list items)

File: submissions/Roophee/Popup_checkbox/index.html (hunk 8 above)
Comment (A-Ostrovnyy) [reply submissions/Roophee/Popup_checkbox/index.html]: Using sprites means you have **one** file with couple of images/icons and you used it in different places, In your case, you have "a sprite" with one icon in every list item.

**Comment by Roophee** [comment]
So, mentors, is my work accepted?

File: submissions/Roophee/Popup_checkbox/index.html (hunk 8 above)
Comment (Roophee) [reply submissions/Roophee/Popup_checkbox/index.html]: I have a promlem. When I move all "sprites" in one file. The SVG icons were gone from my popup and when I call them by id in svg>use nothing happens.

Comment (A-Ostrovnyy) [reply submissions/Roophee/Popup_checkbox/index.html]: [Read more about it](http://dreamhelg.ru/2017/02/symbol-svg-sprite-detail-guide/)
If you don't want to fix it let me know I'll merge your work.

Comment (Roophee) [reply submissions/Roophee/Popup_checkbox/index.html]: Alexandr, merge my pull*. Thanks.

**Review by A-Ostrovnyy — APPROVED** [review]

File: submissions/Roophee/Popup_checkbox/css/main.css (hunk 2 above)
Comment (Roophee) [reply submissions/Roophee/Popup_checkbox/css/main.css:835]: Fixed problems, added new lines, rename files. Make new review, please. Thanks.

**Review by lenadgit — APPROVED** [review]

**Pull request merged**